### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by ionut-arm**
This commit improves some of our documentation and allows the crate docs
to be built for docs.rs.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,3 +24,9 @@ env_logger = "0.7.1"
 [build-dependencies]
 bindgen = "0.52.0"
 pkg-config = "0.3.17"
+
+[features]
+docs = []
```
Comment (hug-dev) [inline Cargo.toml]: Maybe we will also have to add some of thse dependencies as optional if they do not build on the docs.rs server.

Comment (ionut-arm) [reply Cargo.toml]: ... hopefully they do

**Comment by hug-dev** [comment]
Clever fix to allow to have nice docs on docs.rs 😃 In the future we might want to load the library dynamically.

**Comment by ionut-arm** [comment]
> Clever fix to allow to have nice docs on docs.rs 😃 In the future we might want to load the library dynamically.

Agreed, it would've been too much work not justified for now, though

**Pull request merged**